File: src/lib/pjecalc/core/comum/exceptions/negocio-exception.ts
/**
 * PJe-Calc v2.15.1 — NegocioException
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.exceptions.NegocioException
 *
 * Exceção de negócio: RuntimeException que carrega uma lista de
 * MensagemDeRecurso (geralmente falhas de validação).
 */
import type { MensagemDeRecurso } from '../mensagem-de-recurso';
import type { Mensagens } from '../mensagens';

export class NegocioException extends Error {
  private readonly mensagensDeRecurso: MensagemDeRecurso[] = [];

  constructor(causaOrMensagem?: Error | MensagemDeRecurso | null, mensagemDeRecurso?: MensagemDeRecurso | null) {
    super();
    this.name = 'NegocioException';
    if (causaOrMensagem instanceof Error) {
      if (causaOrMensagem.stack) this.stack = causaOrMensagem.stack;
      if (mensagemDeRecurso) this.adicionarMensagemDeRecurso(mensagemDeRecurso);
    } else if (causaOrMensagem) {
      this.adicionarMensagemDeRecurso(causaOrMensagem);
    }
  }

  getMensagensDeRecurso(): MensagemDeRecurso[] { return this.mensagensDeRecurso; }

  adicionarMensagemDeRecurso(mensagem: MensagemDeRecurso): void {
    this.mensagensDeRecurso.push(mensagem);
  }

  agregarExcecao(ne: NegocioException): void {
    for (const m of ne.getMensagensDeRecurso()) {
      this.adicionarMensagemDeRecurso(m);
    }
  }

  existeMensagensDeRecurso(): boolean {
    return this.mensagensDeRecurso.length > 0;
  }

  obterMensagemDeRecursoParaOCampo(mensagem: Mensagens, campo: string | null): MensagemDeRecurso | null {
    for (const msg of this.mensagensDeRecurso) {
      const msgStr = msg.getMensagem();
      if (msgStr === null || msgStr !== String(mensagem) || campo === null) continue;
      if (msg.getAtributo() === campo) return msg;
    }
    return null;
  }
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/exceptions/NegocioException.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.comum.exceptions;

import br.jus.trt8.pjecalc.negocio.comum.MensagemDeRecurso;
import br.jus.trt8.pjecalc.negocio.constantes.Mensagens;
import java.util.ArrayList;
import java.util.List;

public class NegocioException
extends RuntimeException {
    private static final long serialVersionUID = -4411392816296129035L;
    private final List<MensagemDeRecurso> mensagensDeRecurso = new ArrayList<MensagemDeRecurso>();

    public NegocioException(Throwable causa, MensagemDeRecurso mensagemDeRecurso) {
        super(causa);
        if (mensagemDeRecurso != null) {
            this.adicionarMensagemDeRecurso(mensagemDeRecurso);
        }
    }

    public NegocioException(MensagemDeRecurso mensagemDeRecurso) {
        this(null, mensagemDeRecurso);
    }

    public NegocioException() {
        this((MensagemDeRecurso)null);
    }

    public List<MensagemDeRecurso> getMensagensDeRecurso() {
        return this.mensagensDeRecurso;
    }

    public void adicionarMensagemDeRecurso(MensagemDeRecurso mensagemDeRecurso) {
        this.mensagensDeRecurso.add(mensagemDeRecurso);
    }

    public void agregarExcecao(NegocioException ne) {
        for (MensagemDeRecurso mensagem : ne.getMensagensDeRecurso()) {
            this.adicionarMensagemDeRecurso(mensagem);
        }
    }

    public boolean existeMensagensDeRecurso() {
        return !this.mensagensDeRecurso.isEmpty();
    }

    public MensagemDeRecurso obterMensagemDeRecursoParaOCampo(Mensagens mensagem, String campo) {
        for (MensagemDeRecurso msg : this.mensagensDeRecurso) {
            if (!msg.getMensagem().equals(mensagem.name()) || campo == null || !msg.getAtributo().equals(campo)) continue;
            return msg;
        }
        return null;
    }
}

